middleware/logger/config.go

```diff
@@ -43,6 +44,13 @@ type Config struct {
 	// Optional. Default: defaultLogger
 	LoggerFunc func(c fiber.Ctx, data *Data, cfg Config) error
 
+	// Logger allows the use of a custom logger that implements the AllLogger interface.
+	// This field can be used in the LoggerFunc to do the logging.
+	// If you don't define LoggerFunc, it'll use default logger of Fiber without using this field.
+	//
+	// Optional. Default: nil
+	Logger fiberlog.AllLogger
```

Comment:
where is this used in the middleware code ?

Reply:
@haochunchang @efectn 
I think I understand, instead of the buffer in the default logger function another logger is used

three remarks
1. would not add logger to the config if it is only used in the custom variant of the consumer, then it can also use the logger instance from outside, because the logger is not relevant for the existing default logging code
2. if we get the logger instance, we should also use it in the default code, regardless of whether the custom logger function is set or not
3. what is the difference between the logger and the output property? both define the target
maybe you can use output for this?

what do you think about this?

Reply:
I agree that we should also use the logger instance in the default code.

But I haven’t figured out how to use the methods of the AllLogger interface to achieve the same functionality of the original default logger. Especially in the part: https://github.com/gofiber/fiber/blob/c86c3c0a30a87eba3ada305fc70162ea8c875a54/middleware/logger/default_logger.go#L109-L120